account_invoice_pricelist/models/account_move.py

```diff
@@ -0,0 +1,238 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    pricelist_id = fields.Many2one(
+        comodel_name="product.pricelist",
+        string="Pricelist",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+
+    @api.constrains("pricelist_id", "currency_id")
+    def _check_currency(self):
+        for sel in self.filtered(lambda a: a.pricelist_id and a.is_invoice()):
```

Comment:
```suggestion
        for sel in self.filtered("pricelist_id"):
```
IMHO, what you did is making sense, even for non-invoice. Even for Journal Entry, we won't show this pricelist. But when it does, we should block it too. WDYT?

Reply:
I would leave it as it is an extra check, but let me know if you think it's better the way you've suggested and I'll apply the changes.